### PR TITLE
Openthread upmerge to `5e15b93`

### PR DESCRIPTION
**Pull request by canisLupus1313**
Update openthread plus implement dependencies.

### CODE REVIEW

File: modules/openthread/platform/crypto_psa.c [hunk 1]
```diff
@@ -563,4 +563,104 @@ otError otPlatCryptoEcdsaVerify(const otPlatCryptoEcdsaPublicKey *aPublicKey,
 	return psaToOtError(status);
 }
 
+otError otPlatCryptoEcdsaSignUsingKeyRef(otCryptoKeyRef                aKeyRef,
+                                         const otPlatCryptoSha256Hash *aHash,
+                                         otPlatCryptoEcdsaSignature   *aSignature)
+{
+	psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
```
Comment (Damian-Nordic) [inline modules/openthread/platform/crypto_psa.c:570]: The attributes are unused in this function.

File: modules/openthread/platform/crypto_psa.c [hunk 2]
```diff
@@ -563,4 +563,104 @@ otError otPlatCryptoEcdsaVerify(const otPlatCryptoEcdsaPublicKey *aPublicKey,
...
+	psa_reset_key_attributes(&attributes);
+
+	return psaToOtError(status);
+}
+
+otError otPlatCryptoEcdsaVerifyUsingKeyRef(otCryptoKeyRef                    aKeyRef,
+                                           const otPlatCryptoSha256Hash     *aHash,
+                                           const otPlatCryptoEcdsaSignature *aSignature)
+{
+	psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
```
Comment (Damian-Nordic) [inline modules/openthread/platform/crypto_psa.c:596]: unused here as well

File: modules/openthread/platform/crypto_psa.c [hunk 3]
```diff
@@ -563,4 +563,104 @@ otError otPlatCryptoEcdsaVerify(const otPlatCryptoEcdsaPublicKey *aPublicKey,
...
+	psa_set_key_type(&attributes, PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1));
+	psa_set_key_bits(&attributes, 256);
+
+	status = psa_sign_hash(aKeyRef, PSA_ALG_DETERMINISTIC_ECDSA(PSA_ALG_SHA_256), aHash->m8,
+			       OT_CRYPTO_SHA256_HASH_SIZE, aSignature->m8,
+			       OT_CRYPTO_ECDSA_SIGNATURE_SIZE, &signature_length);
+	if (status != PSA_SUCCESS) {
+		goto out;
+	}
+
```
Comment (Damian-Nordic) [inline modules/openthread/platform/crypto_psa.c:585]: maybe assert that `signature_length == OT_CRYPTO_ECDSA_SIGNATURE_SIZE`?

File: modules/openthread/platform/crypto_psa.c [hunk 4]
```diff
@@ -563,4 +563,104 @@ otError otPlatCryptoEcdsaVerify(const otPlatCryptoEcdsaPublicKey *aPublicKey,
...
+out:
+	psa_reset_key_attributes(&attributes);
+
+	return psaToOtError(status);
+}
+
+otError otPlatCryptoEcdsaExportPublicKey(otCryptoKeyRef aKeyRef, otPlatCryptoEcdsaPublicKey *aPublicKey)
+{
+	psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
```
Comment (Damian-Nordic) [inline modules/openthread/platform/crypto_psa.c:619]: unused here as well

File: modules/openthread/platform/crypto_psa.c [hunk 5]
```diff
@@ -563,4 +563,104 @@ otError otPlatCryptoEcdsaVerify(const otPlatCryptoEcdsaPublicKey *aPublicKey,
...
+	psa_reset_key_attributes(&attributes);
+
+	return psaToOtError(status);
+}
+
+otError otPlatCryptoEcdsaGenerateAndImportKey(otCryptoKeyRef aKeyRef)
+{
+	psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
+	psa_status_t status;
+	psa_key_id_t key_id = (psa_key_id_t)aKeyRef; // to samo??
```
Comment (Damian-Nordic) [inline modules/openthread/platform/crypto_psa.c:646]: temporary comment?

File: modules/openthread/platform/crypto_psa.c [hunk 6]
```diff
@@ -563,4 +563,104 @@ otError otPlatCryptoEcdsaVerify(const otPlatCryptoEcdsaPublicKey *aPublicKey,
...
+
+	psa_set_key_usage_flags(&attributes, PSA_KEY_USAGE_EXPORT);
+	psa_set_key_algorithm(&attributes, PSA_ALG_DETERMINISTIC_ECDSA(PSA_ALG_SHA_256));
+	psa_set_key_type(&attributes, PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1));
+	psa_set_key_lifetime(&attributes, PSA_KEY_PERSISTENCE_DEFAULT | PSA_KEY_LOCATION_LOCAL_STORAGE);
```
Comment (Damian-Nordic) [inline modules/openthread/platform/crypto_psa.c:651]:
```suggestion
	psa_set_key_lifetime(&attributes, PSA_KEY_LIFETIME_PERSISTENT);
```

Comment (canisLupus1313) [reply modules/openthread/platform/crypto_psa.c]: fixed.

**Review by Damian-Nordic — DISMISSED** [review]

File: west.yml [hunk 7]
```diff
@@ -210,7 +210,7 @@ manifest:
       revision: 233fb29c52ffa7733ba132a2b5987a8201ba8ec6
       path: modules/lib/open-amp
     - name: openthread
-      revision: 25942910ab57b32f73d9c56d8c6cf07c7cf98c49
+      revision: pull/91/head
```
Comment (rlubos) [inline west.yml:213]: Please update SHA: `1203ea6426a906d62a3921b6d4c6a4a6c8e0abde`

Comment (canisLupus1313) [reply west.yml]: updated

**Review by Damian-Nordic — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

**Pull request merged**